astropy/nddata/utils.py

```diff
@@ -73,11 +150,26 @@ def extract_array(array_large, shape, position):
     position : tuple
         Position of the small array's center, with respect to the large array.
         Coordinates should be in the same order as the array shape.
+    mode : ['partial', 'trim', 'strict']
+        In "partial" and "trim" mode, a partial overlap of the small and the large
+        array is sufficient. In the "strict" mode, the small array has to be
+        fully contained in the large array, otherwise an
+        `~astropy.nddata.utils.PartialOverlapError` is raised. In both modes,
+        non-overlapping arrays will raise a `~astropy.nddata.utils.NoOverlapError`.
+        In "partial" mode, positions in the extracted array, that do not overlap
+        with the original array, will be filled with ``fill_value``. In "trim" mode
+        only the overlapping elements are returned, thus the resulting array may
+        be smaller than requested.
+
+    fill_value : object of type array_large.dtype
+        In "partial" mode ``fill_value`` set the values in the extracted array that
+        do not overlap with ``large_array``.
 
     Returns
     -------
     array_small : `~numpy.ndarray`
-        The extracted array
+        The extracted array. Values that do not overlap with ``array_large``
+        are masked.
```

Comment:
Instead of "are masked", maybe "are handled as described in mode"? Or just revert back to "The extracted array" and leave it at that?